packages/monitor-v2/src/monitor-og/MonitorSnapshotProposals.ts

```diff
@@ -0,0 +1,204 @@
+        id
+        ipfs
+        type
+        choices
+        start
+        end
+        state
+        space {
+          id
+        }
+        scores
+        quorum
+        scores_total
+        plugins
+      }
+    }
+  `);
+
+  // If the GraphQL request fails for any reason, we return an Error object that will be logged by the bot.
+  try {
+    const graphqlData = await retry(
+      () => request<GraphqlData, { plugin: string }>(url, query, { plugin }),
+      retryOptions
+    );
+    // Filter only for proposals that have a properly configured safeSnap or oSnap plugin.
+    return graphqlData.proposals.filter(isSnapshotProposalGraphql);
+  } catch (error) {
+    assert(error instanceof Error, "Unexpected Error type!");
+    return error;
+  }
+};
+
+// Get all active safeSnap/oSnap proposals from all spaces targeting monitored safes (returned in safeSnap format).
+const getActiveSnapshotProposals = async (
+  logger: typeof Logger,
+  params: MonitoringParams
+): Promise<Array<SnapshotProposalGraphql>> => {
+  // Get all active safeSnap/oSnap proposals from all spaces.
+  const snapshotProposals = (
+    await Promise.all(
+      ["oSnap", "safeSnap"].map(async (plugin) =>
+        getActivePluginProposals(plugin, params.graphqlEndpoint, params.retryOptions)
+      )
+    )
+  ).flat();
+
+  // Log all errors that occurred when fetching Snapshot proposals and filter them out.
+  const nonErrorProposals = snapshotProposals.filter((proposal) => {
+    if (!(proposal instanceof Error)) return true;
+    logger.error({
+      at: "oSnapAutomation",
+      message: "Server error when fetching Snapshot proposals",
+      mrkdwn: "Failed to fetch Snapshot proposals",
+      error: proposal,
+      notificationPath: "optimistic-governor",
+    });
+    return false;
+  }) as SnapshotProposalGraphql[];
+
+  // Filter out proposals that do not target any of the monitored safes.
+  return nonErrorProposals.filter((proposal) => {
+    return params.ogAddresses.some((ogAddress) => {
+      const safeSnapPlugin = translateToSafeSnap(proposal.plugins);
+      return safeSnapPlugin.safeSnap.safes.some((safe) => isMatchingSafe(safe, params.chainId, ogAddress));
+    });
+  });
+};
+
+// Returns all notified proposals from saved state (Google Datastore or local file).
+const getNotifiedProposals = async (params: MonitoringParams): Promise<string[]> => {
+  if (params.storage === "datastore") {
+    const datastore = getDatastoreInstance();
+    const query = datastore.createQuery("NotifiedSnapshotProposals");
+    const [notifiedProposals] = await datastore.runQuery(query);
+    return notifiedProposals.map((proposal) => proposal.id);
+  } else {
+    let notifiedProposalsStringified: string;
+    try {
+      const filePath = "./notifiedSnapshotProposals.json";
+      notifiedProposalsStringified = await fsPromises.readFile(filePath, "utf-8");
+    } catch (error) {
+      if ((error as NodeJS.ErrnoException).code === "ENOENT") return [];
```

Comment:
why not check if the file exists, or create the file on first load if this option is selected. then you dont need to check for this specific error since it will always exist

Reply:
agree, simplified file access by first checking file existence